jwst/pathloss/path_loss.py

```diff
@@ -217,50 +258,60 @@ def do_correction(input_model, pathloss_model):
-                slit.pathloss_uniformsource = pathloss_uniform_vector
-                slit.wavelength_uniformsource = wavelength_uniformsource
-                #
-                # Create the 2-d pathloss arrays
-                wavelength_array = slit.wavelength
-                pathloss_pointsource_2d = interpolate_onto_grid(wavelength_array, wavelength_pointsource,
-                                                                pathloss_pointsource_vector)
-                pathloss_uniformsource_2d = interpolate_onto_grid(wavelength_array, wavelength_uniformsource,
-                                                                  pathloss_uniform_vector)
-                slit.pathloss_pointsource2d = pathloss_pointsource_2d
-                slit.pathloss_uniformsource2d = pathloss_uniformsource_2d
+                (wavelength_pointsource,
+                 pathloss_pointsource_vector,
+                 is_inside_slit) = calculate_pathloss_vector(aperture.pointsource_data,
+                                                             aperture.pointsource_wcs,
+                                                             xcenter, ycenter)
+                (wavelength_uniformsource,
+                 pathloss_uniform_vector,
+                 dummy) = calculate_pathloss_vector(aperture.uniform_data,
+                                                    aperture.uniform_wcs,
+                                                    xcenter, ycenter)
+                if is_inside_slit:
+                    #
+                    # Wavelengths in the reference file are in meters, need them to be
+                    # in microns
+                    wavelength_pointsource *= 1.0e6
+                    wavelength_uniformsource *= 1.0e6
+                    slit.pathloss_pointsource = pathloss_pointsource_vector
+                    slit.wavelength_pointsource =  wavelength_pointsource
+                    slit.pathloss_uniformsource = pathloss_uniform_vector
+                    slit.wavelength_uniformsource = wavelength_uniformsource
+                    #
+                    # Create the 2-d pathloss arrays
+                    wavelength_array = slit.wavelength
+                    pathloss_pointsource_2d = interpolate_onto_grid(wavelength_array,
+                                                                    wavelength_pointsource,
+                                                                    pathloss_pointsource_vector)
+                    pathloss_uniformsource_2d = interpolate_onto_grid(wavelength_array,
+                                                                      wavelength_uniformsource,
+                                                                      pathloss_uniform_vector)
+                    slit.pathloss_pointsource2d = pathloss_pointsource_2d
+                    slit.pathloss_uniformsource2d = pathloss_uniformsource_2d
+                else:
+                    log.warning("Source is outside slit")
```

Comment:
May want to increase the warning message to indicate that no correction is computed/applied for this slit.